resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ProjectPreferences.java

```diff
@@ -198,7 +197,7 @@ private static Properties loadProperties(IFile file) throws BackingStoreExceptio
 			Policy.debug("Loading preferences from file: " + file.getFullPath()); //$NON-NLS-1$
 		Properties result = new Properties();
 		try (
-			InputStream input = new BufferedInputStream(file.getContents(true));
+				ByteArrayInputStream input = new ByteArrayInputStream(file.readAllBytes());
```

Comment:
Doesn't this increase the memory consumption and GC pressure noticeably if the preference files are larger here and below? For project files it's probably not a big problem since they are usually relatively small, but preference files are often larger.

Reply:
I am not aware of any preference file that has any significant size. Especially it is just a key value store that is completely read into memory anyway.  For the GC it's easier to clean up a single array that has the exact needed size instead of heaving to cleaning up multiple temp Buffers.

Reply:
FWIW, the biggest org.eclipse.jdt.core.prefs files that I see in my project are slightly less than 40k bytes. It's more a data point, I don't know if it qualifies as significant or not in this context.

Reply:
But realistically, there is nothing in the API that discourages a .prefs file to be 100MB, so assuming all those .prefs files are slim seems too optimistic to me. I support @HannesWell 's impression that unless we have full control or know constraints about the file, going for a stream is safer.
Maybe there is a way to determine the best strategy according to the file size?

Reply:
I agree with @HannesWell here that this looks strange, why should it be better to first read all bytes from a file/stream, then wrap then into a stream and then read that stream again?

e.g. `readAllBytes` of java NIO is effectively creating a buffer array with the size of the file and reads it then from the stream, the same as `BufferedInputStream` would do, so choosing a suitable buffer size here would have the same effect.

Reply:
I removed that example from commits to get progress here.